crates/kitsune_p2p/types/src/tx2/framed.rs

```diff
@@ -382,4 +384,31 @@ mod tests {
 
         wt.await.unwrap().unwrap();
     }
+
+    #[tokio::test]
+    #[cfg(feature = "test_utils")]
+    async fn test_mock_framed() {
+        let mut f = MockAsFramedReader::new();
+        f.expect_read().returning(|_t| {
+            async move {
+                let mut buf = PoolBuf::new();
+                buf.extend_from_slice(b"test");
+                Ok((0.into(), buf))
+            }
+            .boxed()
+        });
+        let (_, buf) = f.read(KitsuneTimeout::from_millis(100)).await.unwrap();
+        assert_eq!(b"test", buf.as_ref());
+
+        let mut f = MockAsFramedWriter::new();
+        f.expect_write().returning(|_, buf, _| {
+            assert_eq!(b"test2", buf.as_ref());
+            async move { Ok(()) }.boxed()
+        });
```

Comment:
FYI I think this is how mockall wants it done. It works in this case since there is only one expectation, but when you've got multiple expectations that can be called in any order, the panic would prevent the other cases from being attempted.

```suggestion
        f
        .expect_write()
        .withf(|_, buf, _| b"test2" == buf.as_ref())
        .returning(|_, buf, _| {
            async move { Ok(()) }.boxed()
        });
```

Reply:
This is part of the test case... if that handler ever receives something other than `b"test2"` then there is a logic error and the test should not pass.

Reply:
Yeah, you don't need to change this, just pointing out "the way mockall wants it done" as an FYI. My suggestion will also cause the test to panic if the handler receives anything other than "btest2", and also will allow for other expectations to be added...which of course won't ever happen in this test case but just pointing it out for potential futures because I learned this recently.

Reply:
ahh, gottcha, yeah, I misunderstood the mockall api, I was assuming it would ignore any other values added...